### ISSUE TITLE
Android port

### ISSUE THREAD

**Issue by DearFox** [OpenStarbound/OpenStarbound#143]
I don't know how crazy difficult it is or how many years from now someone will want to do it - but I would love to be able to play this beautiful game on phone.
Feel free to discuss it here, maybe someone wants to experiment or something like that.

**Comment by hyenatown** [comment]
Yeah, that's understandable. We don't yet have the official controller support CF claimed to be looking into though. I would say if that happens you would be better off playing it that way, but on a long enough timeline anything is possible.

The other thing is dealing with Android's application requirements, not just what CPU the software would run on.
To say Android is fun to develop for is to say the least - a very hard sell. It would take someone who's already familiar with Android to get the ball rolling, but I already do not have a good outlook on device support.
Starbound on Android would require a lot of compromises to how the game functions, including in-game scripts (which oSB doesn't have any of - they're game assets).

**Comment by hyenatown** [comment]
OpenStarbound has a limited scope to what it's trying to accomplish.

- Prolong the shelf life of the game past official support so people can play it for years to come.
- Extend the functionality of the game's internals, which allow for more advanced mods to be made.
- Fix outstanding issues with functionality or performance.

**Comment by lotigara** [comment]
I'm currently working on Android port of this, you can see the progress on <http://git.lotigara.ru/OpenStarbound-android.git>

**Comment by lotigara** [comment]
And there is another one: <https://github.com/RohanBhattacharyya/oSBM>. However, I wasn't able to launch it on my phone.

**Comment by lotigara** [comment]
A technical question: how should be touch controls support implemented, that is, should I use either Starbound's native window system or ImGUI?
In the port above, it is implemented using ImGUI. But I don't like this approach because it feels foreign.

**Comment by grbr404** [comment]
You can get the touch input from SDL, I believe, and then just use the input system.

You should definitely not use ImGUI, as it was added to the codebase explicitly with the notice not to use it for user-facing UI. ImGUI was added for developers only. You can read about that in the changelog of the 0.1.14 release.

So in other words, just use Starbound's native window system.

**Comment by lotigara** [comment]
> You can get the touch input from SDL, I believe, and then just use the input system.
> 
> You should definitely not use ImGUI, as it was added to the codebase explicitly with the notice not to use it for user-facing UI. ImGUI was added for developers only. You can read about that in the changelog of the 0.1.14 release.
> 
> So in other words, just use Starbound's native window system.

Okay, thank you

**Comment by lotigara** [comment]
> You can get the touch input from SDL, I believe, and then just use the input system.
> 
> You should definitely not use ImGUI, as it was added to the codebase explicitly with the notice not to use it for user-facing UI. ImGUI was added for developers only. You can read about that in the changelog of the 0.1.14 release.
> 
> So in other words, just use Starbound's native window system.

How should I exactly do that?

**Comment by Sunbrella114514** [comment]
There is an actively maintained [Android branch ](https://github.com/RohanBhattacharyya/oSBM)that is working well.
